buildSrc/src/test/java/org/elasticsearch/gradle/JdkDownloadPluginTests.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle;
+
+import groovy.lang.Closure;
+import org.elasticsearch.gradle.test.GradleUnitTestCase;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.plugins.ExtraPropertiesExtension;
+import org.gradle.testfixtures.ProjectBuilder;
+import org.junit.BeforeClass;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class JdkDownloadPluginTests extends GradleUnitTestCase {
+    private static Project rootProject;
+
+    @BeforeClass
+    public static void setupRoot() {
+         rootProject = ProjectBuilder.builder().build();
+    }
+
+    public void testNullVersion() {
+        createProject().getTasks().create("example", task -> {
+            assertUsesJdkError(task, null, "linux", "version must be specified to usesJdk");
+        });
+    }
+
+    public void testNullPlatform() {
+        createProject().getTasks().create("example", task -> {
+            assertUsesJdkError(task, "11.0.2+33", null, "platform must be specified to usesJdk");
+        });
+    }
+
+    public void testUnknownPlatform() {
+        createProject().getTasks().create("example", task -> {
+            assertUsesJdkError(task, "11.0.2+33", "unknown", "platform must be one of [linux, windows, darwin]");
+        });
+    }
+
+    public void testJdkAlreadySet() {
+        createProject().getTasks().create("example", task -> {
+            usesJdk(task, "11.0.2+33", "linux");
+            assertUsesJdkError(task, "11.0.2+33", "linux", "jdk version already set for task");
+        });
+    }
+
+    public void testBadVersionFormat() {
+        createProject().getTasks().create("example", task -> {
+            assertUsesJdkError(task, "badversion", "linux", "Malformed jdk version [badversion]");
+        });
+    }
+
+    public void testReuseAcrossTasks() {
+        Project project = createProject();
+        project.getTasks().create("example1", task -> usesJdk(task, "11.0.2+33", "linux"));
+        int numTasks = rootProject.getTasks().size();
+        project.getTasks().create("example2", task -> usesJdk(task, "11.0.2+33", "linux"));
+        assertThat(rootProject.getTasks().size(), equalTo(numTasks));
```

Comment:
how come this isn't off by one  due to the one additional task created on the line above ?